drivers/net/telnet.c

```diff
@@ -1281,7 +1281,8 @@ static int telnet_io_main(int argc, FAR char** argv)
               (CONFIG_TELNET_RXBUFFER_SIZE -
                priv->td_pending - priv->td_offset) > 0)
             {
-              priv->td_fds.sem     = &g_iosem;
+              priv->td_fds.cbargs  = &g_iosem;
```

Comment:
could we let telnet call nx_poll api instead psock_poll

Reply:
So i need add a net_dup() api to get the fd based on the psock and use this fd in telnet_io_main() ?

Reply:
let me remove the poll from telnet driver, so you don't need expose it out side poll.

Reply:
OK

Reply:
here is the patch: https://github.com/apache/incubator-nuttx/pull/7210